### PR TITLE
Importing JoshiaSiegel GitHub Actions

### PR DESCRIPTION
**Pull request by emvaldes**
This Pull Request (PR) is intended only as a placeholder for importing the remote GitHub Actions developed by [Mr. Josiah Siegel](https://github.com/JosiahSiegel). We will work on completing this import aiming to do not case any impact on the existing workflows.

Note: The objective is to remove any dependencies on external repositories outside the DevSecOps team.

### CODE REVIEW

File: .github/actions/action-connect-ovpn/.github/workflows/check_vpn.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install Open VPN
+        run: sudo apt-get install openvpn
+      - name: Connect VPN
+        uses: ./
+        id: connect_vpn
+        with:
+          PING_URL: '${{ secrets.PING_URL }}'
```
Comment (devopsmatt) [inline .github/actions/action-connect-ovpn/.github/workflows/check_vpn.yml:16]: Secret

File: .github/actions/action-connect-ovpn/action.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    - name: Install OpenVPN
+      run: |
+        sudo apt-get update
+        sudo apt-get install openvpn
+        sudo apt-get install openvpn-systemd-resolved
+      shell: bash
+
+    - name: Connect VPN
+      env:
+        TLS_KEY: ${{ inputs.TLS_KEY }}
```
Comment (devopsmatt) [inline .github/actions/action-connect-ovpn/action.yml:39]: validate inputs and env vars are set

File: .github/actions/runleaks/.github/workflows/scan_public.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+name: Scan Public Logs
+
+on: 
+  workflow_dispatch:
+    inputs:
+      repos:
+        description: 'Repos to scan:'
+        required: true
+        default: '[\"josiahsiegel\/runleaks\",\"josiahsiegel\/AzViz-action\"]'
```
Comment (devopsmatt) [inline .github/actions/runleaks/.github/workflows/scan_public.yml:9]: update repo

File: .github/actions/runleaks/.github/workflows/scan_public.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    outputs:
+      repos: ${{ env.REPOS }}
+    steps:
+      - name: Set workflow_dispatch matrix
+        if: github.event_name == 'workflow_dispatch'
+        run: echo "REPOS=${{ github.event.inputs.repos }}" >> $GITHUB_ENV
+
+      - name: Fetch random repo name
+        if: github.event_name == 'schedule' || github.event_name == 'push'
+        uses: JosiahSiegel/randomrepo@v1.1
```
Comment (devopsmatt) [inline .github/actions/runleaks/.github/workflows/scan_public.yml:28]: needs update

File: .github/actions/runleaks/README.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
+# runleaks
+
+[![Scan Action Logs](https://github.com/JosiahSiegel/runleaks/actions/workflows/main.yml/badge.svg?branch=main)](https://github.com/JosiahSiegel/runleaks/actions/workflows/main.yml)
```
Comment (devopsmatt) [inline .github/actions/runleaks/README.md:3]: needs update

File: .github/actions/runleaks/README.md [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    description: 'Count of exceptions'
+```
+
+## Usage
+ * Note: [GitHub rate limits](#rate-limits)
+```yml
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Scan run logs
+        uses: josiahsiegel/runleaks@v1
```
Comment (devopsmatt) [inline .github/actions/runleaks/README.md:59]: needs upddate

File: .github/actions/runleaks/README.md [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+          run-limit: 200
+          min-days-old: 0
+          max-days-old: 4
+          fail-on-leak: true
+```
+
+## Local testing
+  * Registers default patterns
+```sh
+git clone https://github.com/JosiahSiegel/runleaks.git
```
Comment (devopsmatt) [inline .github/actions/runleaks/README.md:106]: needs update

File: .github/actions/terraform-stats/README.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
+# Terraform Stats
+
+[![Test action](https://github.com/JosiahSiegel/terraform-stats/actions/workflows/test_action.yml/badge.svg)](https://github.com/JosiahSiegel/terraform-stats/actions/workflows/test_action.yml)
```
Comment (devopsmatt) [inline .github/actions/terraform-stats/README.md:3]: needs update

File: .github/actions/terraform-stats/README.md [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+   * "Change" refers to change actions that Terraform plans to use to move from the prior state to a new state.
+5. Change percent
+   * Percentage of changes to total resources.
+6. Resource changes
+
+## Usage
+
+```yml
+- name: Terraform stats
+  uses: josiahsiegel/terraform-stats@<latest-version>
```
Comment (devopsmatt) [inline .github/actions/terraform-stats/README.md:22]: needs update

**Review by devopsmatt — COMMENTED** [review]
some minor items to address

**Comment by scott-aquia** [comment]
Inserting a comment here to address the request from the bot

File: .github/actions/action-connect-ovpn/.github/workflows/check_vpn.yml (hunk 1 above)
Comment (emvaldes) [reply .github/actions/action-connect-ovpn/.github/workflows/check_vpn.yml:16]: We do not have this specific secret in this repo. I will continue to review this workflow.

File: .github/actions/action-connect-ovpn/action.yml (hunk 2 above)
Comment (emvaldes) [reply .github/actions/action-connect-ovpn/action.yml:39]: We have this input defined in the Environment Secrets section.

File: .github/actions/action-connect-ovpn/.github/workflows/check_vpn.yml (hunk 1 above)
Comment (emvaldes) [reply .github/actions/action-connect-ovpn/.github/workflows/check_vpn.yml:16]: All these issues are not documented and we are in the process of reaching out to Mr. Josiah Siegel and ask for his assistance in figuring out the different environment and secrets we need to setup in this project.

File: .github/actions/action-connect-ovpn/action.yml (hunk 2 above)
Comment (emvaldes) [reply .github/actions/action-connect-ovpn/action.yml:39]: All these issues are not documented and we are in the process of reaching out to Mr. Josiah Siegel and ask for his assistance in figuring out the different environment and secrets we need to setup in this project.

File: .github/actions/runleaks/.github/workflows/scan_public.yml (hunk 3 above)
Comment (emvaldes) [reply .github/actions/runleaks/.github/workflows/scan_public.yml]: All these issues are not documented and we are in the process of reaching out to Mr. Josiah Siegel and ask for his assistance in figuring out the different environment and secrets we need to setup in this project.

File: .github/actions/runleaks/README.md (hunk 5 above)
Comment (emvaldes) [reply .github/actions/runleaks/README.md]: All these issues are not documented and we are in the process of reaching out to Mr. Josiah Siegel and ask for his assistance in figuring out the different environment and secrets we need to setup in this project.

File: .github/actions/terraform-stats/README.md (hunk 9 above)
Comment (emvaldes) [reply .github/actions/terraform-stats/README.md]: All these issues are not documented and we are in the process of reaching out to Mr. Josiah Siegel and ask for his assistance in figuring out the different environment and secrets we need to setup in this project.

File: .github/actions/runleaks/README.md (hunk 7 above)
Comment (emvaldes) [reply .github/actions/runleaks/README.md]: All these issues are not documented and we are in the process of reaching out to Mr. Josiah Siegel and ask for his assistance in figuring out the different environment and secrets we need to setup in this project.

File: .github/actions/runleaks/README.md (hunk 6 above)
Comment (emvaldes) [reply .github/actions/runleaks/README.md]: All these issues are not documented and we are in the process of reaching out to Mr. Josiah Siegel and ask for his assistance in figuring out the different environment and secrets we need to setup in this project.

File: .github/actions/runleaks/.github/workflows/scan_public.yml (hunk 4 above)
Comment (emvaldes) [reply .github/actions/runleaks/.github/workflows/scan_public.yml]: All these issues are not documented and we are in the process of reaching out to Mr. Josiah Siegel and ask for his assistance in figuring out the different environment and secrets we need to setup in this project.

File: .github/actions/deploy-backend/action.yml [hunk 10]
```diff
@@ -333,7 +332,10 @@ runs:
 
     - name: Validate function app checksum
       if: inputs.checksum-validation == 'true'
+
       uses: JosiahSiegel/checksum-validate-action@ebdf8c12c00912d18de93c483b935d51582f9236
+      ## emvaldes@hotmail.com (Replace) uses: ./.github/actions/checksum-validate-action
```
Comment (devopsmatt) [inline .github/actions/deploy-backend/action.yml:337]: corp email please

File: .github/actions/vpn-azure/action.yml [hunk 11]
```diff
@@ -41,6 +41,8 @@ runs:
       shell: bash
 
     - uses: josiahsiegel/action-connect-ovpn@794339aff94452216c97f609476c367a43a31295
+    ## emvaldes@hotmail.com (Replace) - uses: ./.github/actions/action-connect-ovpn
```
Comment (devopsmatt) [inline .github/actions/vpn-azure/action.yml:44]: corp email please

File: .github/dependabot.yml [hunk 12]
```diff
@@ -120,6 +120,12 @@ updates:
     schedule:
       interval: "daily"
 
+  ## Importing Josiah Siegel's external/remote GitHub Actions
```
Comment (devopsmatt) [inline .github/dependabot.yml:123]: no need for comment here

File: .github/vpn/config.ovpn [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+auth-user-pass secret.txt
+cipher AES-256-CBC
+auth SHA256
+comp-lzo no
+route-delay 4
+verb 3
+reneg-sec 0
+ca ca.crt
+cert user.crt
+key user.key
```
Comment (devopsmatt) [inline .github/vpn/config.ovpn:21]: newline at the end please

File: .github/workflows/alert_terraform_changes.yml [hunk 14]
```diff
@@ -26,9 +26,12 @@ jobs:
           user-key: ${{ secrets.USER_KEY }}
           sp-creds: ${{ secrets.SERVICE_PRINCIPAL_CREDS }}
           tf-auth: true
-          
+
       - name: Collect Terraform stats
+
         uses: josiahsiegel/terraform-stats@68b8cbe42c494333fbf6f8d90ac86da1fb69dcc2
+        ## emvaldes@hotmail.com (Replace) - uses: ./.github/actions/terraform-stats
```
Comment (devopsmatt) [inline .github/workflows/alert_terraform_changes.yml:33]: corp email please

File: .github/actions/deploy-backend/action.yml [hunk 15]
```diff
@@ -152,7 +151,7 @@ runs:
       shell: bash
       env:
         SSL_KEY: ${{ steps.key-vault.outputs[env.KEY_NAME] }}
-    
+
```
Comment (emvaldes) [inline .github/actions/deploy-backend/action.yml:154]: The system is randomly adding new lines and I guess it's related to Windows/Linux \n.
This should not be an issue.

File: .github/actions/deploy-backend/action.yml (hunk 10 above)
Comment (emvaldes) [inline .github/actions/deploy-backend/action.yml:337]: The objective here is to provide reference to where we need to change repos that are referenced within the project.

File: .github/actions/action-connect-ovpn/.gitignore [hunk 16]
```diff
@@ -0,0 +1,4 @@
+.env
+*.crt
```
Comment (devopsmatt) [inline .github/actions/action-connect-ovpn/.gitignore:2]: should these be globbed?  `**/*key` etc?

File: .github/workflows/check_vpn.yml [hunk 17]
```diff
@@ -0,0 +1,29 @@
+name: Check connect vpn
```
Comment (devopsmatt) [inline .github/workflows/check_vpn.yml:1]: Not needed here, we can test with the actual workflow.  Please remove before merging.

File: .github/workflows/deploy_terraform.yml [hunk 18]
```diff
@@ -51,7 +51,10 @@ jobs:
           sp-creds: ${{ secrets.SERVICE_PRINCIPAL_CREDS }}
           tf-auth: true
       - name: Collect Terraform stats
+
         uses: josiahsiegel/terraform-stats@68b8cbe42c494333fbf6f8d90ac86da1fb69dcc2
+        ## emvaldes@hotmail.com (Replace) - uses: ./.github/actions/terraform-stats
```
Comment (devopsmatt) [inline .github/workflows/deploy_terraform.yml:56]: corp email please

File: .github/workflows/log_management.yml [hunk 19]
```diff
@@ -12,7 +12,10 @@ jobs:
     steps:
       - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
       - name: Workflow Housekeeper - workflows NOT in default branch
+
         uses: JosiahSiegel/workflow-housekeeper@731cc20bb613208b34efb6ac74aab4ba147abb50
+        ## emvaldes@hotmail.com (Replace) - uses: ./.github/actions/workflow-housekeeper
```
Comment (devopsmatt) [inline .github/workflows/log_management.yml:17]: corp email please

File: .github/workflows/log_management.yml [hunk 20]
```diff
@@ -21,7 +24,10 @@ jobs:
           retain-run-count: 0
           dry-run: false
       - name: Workflow Housekeeper - workflows in default branch
+
         uses: JosiahSiegel/workflow-housekeeper@731cc20bb613208b34efb6ac74aab4ba147abb50
+        ## emvaldes@hotmail.com (Replace) - uses: ./.github/actions/workflow-housekeeper
```
Comment (devopsmatt) [inline .github/workflows/log_management.yml:29]: corp email please

File: .github/workflows/prepare_deployment_branch.yaml [hunk 21]
```diff
@@ -27,13 +27,19 @@ jobs:
           echo "Branch name: \"${BRANCH_NAME}\""
 
       - name: "Create branch '${{ env.BRANCH_NAME }}' to contain the changes for the deployment on ${{ env.DEPLOYMENT_DATE }}"
+
         uses: JosiahSiegel/remote-branch-action@dbe7a2138eb064fbfdb980abee918091a7501fbe
+        ## emvaldes@hotmail.com (Replace) - uses: ./.github/actions/remote-branch-action
```
Comment (devopsmatt) [inline .github/workflows/prepare_deployment_branch.yaml:32]: corp email please

File: .github/workflows/prepare_deployment_branch.yaml [hunk 22]
```diff
@@ -27,13 +27,19 @@ jobs:
...
+
         with:
           branch: "${{ env.BRANCH_NAME }}"
 
       - name: "Prepare a Pull Request from ${{ env.BRANCH_NAME }} into production branch"
         id: pr
+
         uses: JosiahSiegel/reliable-pull-request-action@ae8d0c88126329ee363a35392793d0bc94cb82e7
+        ## emvaldes@hotmail.com (Replace) - uses: ./.github/actions/reliable-pull-request-action
```
Comment (devopsmatt) [inline .github/workflows/prepare_deployment_branch.yaml:41]: corp email please

File: .github/workflows/release_to_azure.yml [hunk 23]
```diff
@@ -144,7 +144,10 @@ jobs:
         env:
           checksum_validation: ${{ vars.CHECKSUM_VALIDATION }}
         if: needs.pre_job.outputs.has_router_change == 'true' && env.checksum_validation == 'true'
+
         uses: JosiahSiegel/checksum-validate-action@ebdf8c12c00912d18de93c483b935d51582f9236
+        ## emvaldes@hotmail.com (Replace) - uses: ./.github/actions/checksum-validate-action
```
Comment (devopsmatt) [inline .github/workflows/release_to_azure.yml:149]: corp email please

File: .github/workflows/scan_action_logs.yml [hunk 24]
```diff
@@ -12,7 +12,10 @@ jobs:
     steps:
       - uses: actions/checkout@d632683dd7b4114ad314bca15554477dd762a938
       - name: Scan run logs
+
         uses: josiahsiegel/runleaks@4dd30d107c03b6ade87978e10c94a77015e488f9
+        ## emvaldes@hotmail.com (Replace) - uses: ./.github/actions/runleaks
```
Comment (devopsmatt) [inline .github/workflows/scan_action_logs.yml:17]: corp email please

**Review by devopsmatt — CHANGES_REQUESTED** [review]
A few minor nitpicks to address.

File: .github/actions/deploy-backend/action.yml (hunk 10 above)
Comment (emvaldes) [reply .github/actions/deploy-backend/action.yml]: I am using this as the contributor's email as it's the one for the account that is registered in GitHub.
Nevertheless, if required I will change it.

File: .github/workflows/scan_action_logs.yml (hunk 24 above)
Comment (emvaldes) [reply .github/workflows/scan_action_logs.yml]: All these GHA that were supposed to be imported are no longer in scope and were removed.
I will leave these markers for future evaluation.

``` console
./.github/actions/runleaks
./.github/actions/terraform-stats
./.github/actions/workflow-housekeeper
./.github/actions/workflow-housekeeper
./.github/actions/remote-branch-action
./.github/actions/reliable-pull-request-action
./.github/actions/checksum-validate-action
./.github/actions/terraform-stats
./.github/actions/checksum-validate-action
```

File: .github/actions/action-connect-ovpn/.gitignore (hunk 16 above)
Comment (emvaldes) [reply .github/actions/action-connect-ovpn/.gitignore:2]: I would agree but eventually these will be merged into the main .gitignore so I am not sure if it's worth to abstract it or just leave it as explicit exclusions.

File: .github/actions/deploy-backend/action.yml (hunk 10 above)
Comment (emvaldes) [reply .github/actions/deploy-backend/action.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/actions/vpn-azure/action.yml (hunk 11 above)
[the remark this answers is not in the document]
Comment (emvaldes) [reply .github/actions/vpn-azure/action.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/dependabot.yml (hunk 12 above)
[the remark this answers is not in the document]
Comment (emvaldes) [reply .github/dependabot.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/alert_terraform_changes.yml (hunk 14 above)
[the remark this answers is not in the document]
Comment (emvaldes) [reply .github/workflows/alert_terraform_changes.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/deploy_terraform.yml (hunk 18 above)
[the remark this answers is not in the document]
Comment (emvaldes) [reply .github/workflows/deploy_terraform.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/log_management.yml (hunk 19 above)
[the remark this answers is not in the document]
Comment (emvaldes) [reply .github/workflows/log_management.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/log_management.yml (hunk 20 above)
[the remark this answers is not in the document]
Comment (emvaldes) [reply .github/workflows/log_management.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/prepare_deployment_branch.yaml (hunk 21 above)
[the remark this answers is not in the document]
Comment (emvaldes) [reply .github/workflows/prepare_deployment_branch.yaml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/prepare_deployment_branch.yaml (hunk 22 above)
[the remark this answers is not in the document]
Comment (emvaldes) [reply .github/workflows/prepare_deployment_branch.yaml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/release_to_azure.yml (hunk 23 above)
[the remark this answers is not in the document]
Comment (emvaldes) [reply .github/workflows/release_to_azure.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/scan_action_logs.yml (hunk 24 above)
[the remark this answers is not in the document]
Comment (emvaldes) [reply .github/workflows/scan_action_logs.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/actions/deploy-backend/action.yml (hunk 10 above)
Comment (emvaldes) [reply .github/actions/deploy-backend/action.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/dependabot.yml (hunk 12 above)
Comment (emvaldes) [reply .github/dependabot.yml]: I will get it removed now.

File: .github/workflows/alert_terraform_changes.yml (hunk 14 above)
Comment (emvaldes) [reply .github/workflows/alert_terraform_changes.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/scan_action_logs.yml (hunk 24 above)
Comment (emvaldes) [reply .github/workflows/scan_action_logs.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/actions/vpn-azure/action.yml (hunk 11 above)
Comment (emvaldes) [reply .github/actions/vpn-azure/action.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/vpn/config.ovpn (hunk 13 above)
Comment (emvaldes) [reply .github/vpn/config.ovpn]: Added the end-of-file new line.

File: .github/workflows/deploy_terraform.yml (hunk 18 above)
Comment (emvaldes) [reply .github/workflows/deploy_terraform.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/log_management.yml (hunk 19 above)
Comment (emvaldes) [reply .github/workflows/log_management.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/prepare_deployment_branch.yaml (hunk 21 above)
Comment (emvaldes) [reply .github/workflows/prepare_deployment_branch.yaml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/prepare_deployment_branch.yaml (hunk 22 above)
Comment (emvaldes) [reply .github/workflows/prepare_deployment_branch.yaml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/release_to_azure.yml (hunk 23 above)
Comment (emvaldes) [reply .github/workflows/release_to_azure.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/log_management.yml (hunk 20 above)
Comment (emvaldes) [reply .github/workflows/log_management.yml]: I have updated the email address to reflect the Aquia corporate email.

File: .github/workflows/check_vpn.yml (hunk 17 above)
Comment (emvaldes) [reply .github/workflows/check_vpn.yml]: Are you suggesting to not include the Check-VPN workflow file now and leave it as part of the integration phase?

File: .github/actions/deploy-backend/action.yml (hunk 15 above)
Comment (devopsmatt) [reply .github/actions/deploy-backend/action.yml:154]: I've set my editor to adde a newline and remove trailing spaces on save.  This is more for consistency than anything else.

**Review by emvaldes — COMMENTED** [review]
Addressing all comments included in the PR.

**Review by devopsmatt — APPROVED** [review]
LGTM

**Pull request merged**